regularizer_new.py

```diff
@@ -38,16 +72,128 @@ def cache(self, input_str, parse_split=None):
             self._cache[input_str] = (slice_dict, masked_strs, input_masks)
         
         return self._cache[input_str]
+    
+    def get_phrases(self, str):
+        # Sample phrases from given string
+        phrases = set()
+
+        # It is guaranteed that str is larger than phrase len
+        for iter in range(self.sample_num):
+            if (self.dataset in ["ds-addmult-mod10", "dyck"]):
+                # ensure phrases are the largest nested subtree
+                if (self.dataset == "dyck"):
+                    str = str.split(" ")
```

Comment:
Variable shadowing is best to avoid when possible.

Reply:
Changed, thanks!